### PR TITLE
[MNT] Remove python 3.12 constraint for deep learners

### PR DESCRIPTION
**Pull request by hadifawaz1999**
trying to remove constraint of python 3.12 for testing deep learners, fixing as well an old issue related to Encoder network removing the need of lamda layer

### CODE REVIEW

**Comment by TonyBagnall** [comment]
good stuff, is there an issue related to the old issue with encoders? If so, could you tag it?

File: aeon/networks/base.py [hunk 1]
```diff
@@ -4,23 +4,18 @@
...
-    _check_soft_dependencies,
-)
+from aeon.utils.validation._dependencies import _check_soft_dependencies
 
 
 class BaseDeepLearningNetwork(ABC):
     """Abstract base class for deep learning networks."""
 
-    def __init__(self, soft_dependencies="tensorflow", python_version="<3.12"):
+    def __init__(self, soft_dependencies="tensorflow"):
```
Comment (MatthewMiddlehurst) [inline aeon/networks/base.py:13]: I would keep this but just default it to `None` or somthing

File: aeon/networks/tests/test_all_networks.py [hunk 2]
```diff
@@ -21,12 +18,10 @@
 def test_network_config(network):
     """Tests if the config dictionary of classes is correctly configured."""
     assert "python_dependencies" in network._config.keys()
-    assert "python_version" in network._config.keys()
```
Comment (MatthewMiddlehurst) [inline aeon/networks/tests/test_all_networks.py:24]: same with all this stuff

Comment (hadifawaz1999) [reply aeon/networks/tests/test_all_networks.py]: done

File: aeon/networks/base.py (hunk 1 above)
Comment (hadifawaz1999) [reply aeon/networks/base.py]: done added <=3.12 by default

**Review by MatthewMiddlehurst — APPROVED** [review]
Made a few changes which look like they were left over. PR looks fine to me now, but feel free to change if i made any mistakes above.

**Comment by hadifawaz1999** [comment]
> Made a few changes which look like they were left over. PR looks fine to me now, but feel free to change if i made any mistakes above.

thanks some i forgot probably to add when reverting the python version, the encoder thing is not actually necessary as it is the same as the one of base class which is why i removed it, but does not hurt

**Comment by hadifawaz1999** [comment]
@MatthewMiddlehurst seems <=3.12 should be <3.13 because we actually use 3.12.7 ! will fix that thanks for the changes CI wouldnt have detected them

**Comment by hadifawaz1999** [comment]
pushed the changes to <3.13 and added the config in every network, even if it inherits the base class config, better to make it clear for each one, to be consistent as AE based ones we re-do the config because the structure element is different

**Review by MatthewMiddlehurst — APPROVED** [review]

**Pull request merged**